Attestation

PavelSproot opened the pull request.

PavelSproot committed:
Подчищаем хвосты.

PavelSproot committed:
Создана папка аттестации (чистая)

PavelSproot committed:
Создана структура, настроен Redux и взаимодействие с API

PavelSproot committed:
Чуть оклемался после Ковида...
Доделал основной функционал первой аттестации.

Ricorder reviewed (commented):
- нет возможности редактировать пользователя;
- невозможно перейти на пользователя, под которым осуществлён вход, ссылка вверху не работает;
- невозможно перейти на пользователя из поста.

PavelSproot committed:
Имя пользователя в шапке и в посте сделал активным. При клике переходит на профиль пользователя.
Редактирование пользователя сделаю чуть позже, не реализовал эту опцию пока - мало времени.

Ricorder reviewed (commented):
- нет возможности редактировать пользователя;

PavelSproot committed:
Редактирование прикрутил.

PavelSproot committed:
Доделал редактированеие пользователя.
Вроде как всё теперь.
(Файлы коммитятся все кучей из проекта т.к. при чекауте почемуто перевод строки в конце каждой строки в каждом файле пропадает, приходится фиксить все файлы).

PavelSproot commented:
Доделал редактирование пользователя.
Вроде как всё теперь.

PavelSproot committed:
Подобрал мелочевку.

Ricorder reviewed (commented):
![Body-not-valid](https://user-images.githubusercontent.com/48915901/147384676-02380c96-1264-4492-951a-dc926fe1cfe6.jpg)

При редактировании сначала было всё хорошо, потом загрузи фото, получилось и пытался сохранить, правда с пустым полем email, но оно по идее должно было быть заполнено предыдущим значением и вот такая ошибка вышла

PavelSproot commented:
Я тут как раз заливал изменения, может из-за этого? Не могу воспроизвести ситуацию... email я вообще блокирую от редактирования т.к. по документации API его нельзя менять... 
Если не трудно, дайте, пожалуйста, последовательность действий.
Та ошибка, что на экране - это когда постится незаполненный пользователь... но как добиться, чтобы он оказался незаполненным, я не могу...

Ricorder reviewed (commented):
Вообще перестало запускаться

PavelSproot commented:
Перепроверил версию - у меня та же самая версия.... И у меня все работает хорошо...
Что можно сделать?
Кстати, в 29-м задании (локализация) - та же самая аттестация... функционал должен быть идентичен. Может там запускается?

Ricorder commented:
![Git](https://user-images.githubusercontent.com/48915901/147388135-bf1640d1-de2f-429c-9a95-12e3b06eef4e.jpg)

Ricorder commented:
![VSCode](https://user-images.githubusercontent.com/48915901/147388163-6b8c569e-0b9a-4f88-87b6-d009a6b84b31.jpg)

Когда запускаю yarn start(работаю через yarn, а не через npm), то выходит следующая ошибка

PavelSproot commented:
Папка 29 пуста, видимо, потомучто Вы уже ее замёрджили вчера - мне об этом письмо пришло :))) и я, следом, ее с develop в main замерджил, я думал, что все нормально, что Вы уже ее проверили т.к. ПуллРеквест закрыли.
Я сейчас вторую папку 29 сделал и залил :) Посмотрите там. Папка homework_29_2 и ветка соответствующая.

PavelSproot commented:
А касательно ошибки.... у меня в ОС браузер по умолчанию не chrome, поэтому пришлось доставить модуль cross-env, чтобы мое веб-приложение в Хроме запускалось по умолчанию... это обычный модуль, он в package.json в моем прописан... можете его доустановить себе и ошибка должна уйти :))

PavelSproot commented:
ну, либо подправить в package.json в разделе scripts команду start.
![image](https://user-images.githubusercontent.com/92236965/147397417-02a7accd-b12c-405b-a797-67812365c44b.png)

Ricorder reviewed (approved):
Хорошо
